tensorflow/core/platform/default/logging.cc

```diff
@@ -31,28 +31,41 @@ limitations under the License.
 #include <string.h>
 #include <time.h>
 
-#include <string>
+#include <algorithm>
+#include <mutex>
+#include <queue>
 #include <unordered_map>
+#include <vector>
 
 namespace tensorflow {
+namespace internal {
 
-void TFAddLogSink(TFLogSink* sink) {
```

Comment:
Please keep the existing API the same (TF{Add,Remove}LogSink). It can forward to your proposed implementation.